### PR TITLE
Maxihost provider

### PR DESCRIPTION
**Pull request by mpempekos**
### Description

Implemented main classes and methods for connecting with Maxihost's API. API ref can be found: https://developers.maxihost.com

### Status

- Done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [x] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [x] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by Kami** [comment]
Thanks for the contribution :+1: 

It looks like a good start.

Since it's a bigger contribution, can you please also sign an ICLA (http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes)?

In addition to that, can you also add some more test cases for various edge cases (invalid credentials, etc.) and corresponding documentation page?

File: libcloud/compute/drivers/maxihost.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        """
+        Create a node.
+
+        :return: The newly created node.
+        :rtype: :class:`Node`
+        """
+        attr = {'hostname': name, 'plan': size.id,
+                'operating_system': image.id,
+                'facility': location.id.lower(), 'billing_cycle': 'monthly',
+                'ex_ssh_key_ids': ex_ssh_key_ids}
```
Comment (Kami) [inline libcloud/compute/drivers/maxihost.py:39]: Looking at the docs (https://developers.maxihost.com/reference#post_devices), the attribute name on the API side should be ``ssh_keys``, right?

File: libcloud/compute/drivers/maxihost.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+        for key in data:
+            extra[key] = data[key]
+
+        node = Node(id=data['id'], name=data['description'], state=state,
+                    private_ips=private_ips, public_ips=public_ips,
+                    driver=self, extra=extra)
+        return node
+
+    def list_locations(self, available=True):
```
Comment (Kami) [inline libcloud/compute/drivers/maxihost.py:122]: ``available`` -> ``ex_available``.

File: libcloud/compute/drivers/maxihost.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        """
+        List all the available SSH keys.
+
+        :return: Available SSH keys.
+        :rtype: ``list`` of :class:`KeyPair`
+        """
+        data = self.connection.request('/account/keys')
+        return list(map(self._to_key_pair, data.object['ssh_keys']))
+
+    def create_key_pair(self, name, public_key=''):
```
Comment (Kami) [inline libcloud/compute/drivers/maxihost.py:193]: Should remove default argument value if the argument is required (aka `(self, name, public_key)`).

**Comment by Kami** [comment]
Thanks for adding the docs, etc.

There are some more small issues left which I will address when merging this into master.

**Comment by Kami** [comment]
I've made small changes as per PR review comments (3a9ef1f2ba89b77f61a7791681edd60661364fc6, 1170e9c3d93276fd20986172acaea863e9e4dd69, 2fe5304b681b81584ece8a0ebfb61e752a6a2fd3), fixed tests (baf956e5cfe47b9b375dfd64cfc30c5a21513ce4) and merged this into trunk.

In the future, more test cases for various edge cases (invalid credentials, API returns an error, etc.) would also be great.

Thanks again for your contribution.

**Comment by mpempekos** [comment]
@Kami   thanks for addressing the issues
